### PR TITLE
Move wormhole concentrator outside pod ruche array

### PR DESCRIPTION
**Pull request by gaozihou**
Move wormhole concentrator outside pod ruche array to accommodate to top-level design infrastructure
SPMD testbench updated accordingly

### CODE REVIEW

**Comment by tommydcjung** [comment]
Are we still merging top vcache and bottom vcache traffics? If that's the case, you would need to set different concentrator IDs.

**Comment by gaozihou** [comment]
I guess we are going to merge top and bottom vcache (4 : 1) using a single concentrator with 2-bits Cid (currently it is using 1-bit). Do you think we should also add this change to the spmd testbench?

**Comment by tommydcjung** [comment]
Yeah I think we should also update the spmd testbench so it matches what we plan to tapeout.
When you are making this change, make sure that the address mapping  in the wormhole test mem does not overlap with the top and bottom vcaches. now that x-src-coord is not unique.

**Comment by gaozihou** [comment]
Got it, thanks for your suggestions Tommy!

File: v/bsg_manycore_tile_vcache_array.v [hunk 1]
```diff
@@ -120,7 +121,7 @@ module bsg_manycore_tile_vcache_array
 
       ,.my_wh_cord_i({pod_x_i, x_subcord_width_lp'(i)})
       ,.dest_wh_cord_i(dest_wh_cord_i)
-      ,.my_wh_cid_i(wh_cid_width_p'(i%wh_ruche_factor_p))
+      ,.my_wh_cid_i(wh_cid_width_p'({i%wh_ruche_factor_p, my_cid_i}))
```
[the remark this answers is not in the document]
Comment (gaozihou) [reply v/bsg_manycore_tile_vcache_array.v:124]: Yeah I can reverse it, in order to support non-power-of-2 ruche-factor the logic will be 
wh_cid_width_p'((i%wh_ruche_factor_p)+(my_cid_i*wh_ruche_factor_p))
Then the nonsynth_mem can determine north and south row by (wh_cid/wh_ruche_factor_p)

[the remark this answers is not in the document]
Comment (gaozihou) [reply v/bsg_manycore_tile_vcache_array.v:124]: I can revert to the previous commit if the reversed cid is better...

**Comment by gaozihou** [comment]
@tommydcjung passing pod_1x1, pod_1x1_hbm2 regression, pod_4x4 regression in progress, should be good to go

**Review by tommydcjung — APPROVED** [review]
Is this ready to merge?

**Comment by gaozihou** [comment]
Yes, it is ready to merge. Done for pod_1x1, pod_1x1_hbm2 spmd regression, done with most of the pod_4x4 spmd regression (finished 108 out of 116 tests, with 8 tests timed out)

**Pull request merged**